### DISCUSSION TITLE
Reading raw partitions on Windows

### DISCUSSION THREAD

**Discussion by God-damnit-all** [BurntSushi/ripgrep#2309] [General]
RecoverPy is a tool I would like to use on Windows, but its dependence on grep makes that difficult for the developer to implement. Sure enough, looking through the documentation, I don't see anything that could accomplish this.

Could ripgrep get a feature that allows it to do this?

**Comment by BurntSushi** [comment]
I have no idea what you're asking. Have you tried using ripgrep? I don't see why grep would be able to do it and ripgrep not, but I don't know Windows. I assume both tools use the standard file reading APIs.

Your question seems to assume I have a lot more Windows knowledge than I actually do. :-)

**Reply by God-damnit-all** [reply]
> I have no idea what you're asking. Have you tried using ripgrep? I don't see why grep would be able to do it and ripgrep not, but I don't know Windows. I assume both tools use the standard file reading APIs.
> 
> Your question seems to assume I have a lot more Windows knowledge than I actually do. :-)

I think it has more to do with how Linux handles partitions, essentially making it possible for grep to read an entire partition as a single, raw file.

**Reply by BurntSushi** [reply]
Once again: *have you actually tried it*? And if not, why not?

I have personally used both ripgrep and grep to search raw partitions on Linux. It works just fine. No extra support needed.

**Reply by God-damnit-all** [reply]
> Once again: _have you actually tried it_? And if not, why not?
> 
> I have personally used both ripgrep and grep to search raw partitions on Linux. It works just fine. No extra support needed.

Well yeah, I'm sure it works on Linux, it's just that Windows doesn't really offer any mechanism to read raw partitions in the first place.

**Reply by BurntSushi** [reply]
Well then we're back to square one. I don't know what you're asking for.

You might need to loop in someone with more Windows expertise here. Or if grep really does work only because of special support added to it, then you might consider finding the relevant code and linking it here.

**Reply by God-damnit-all** [reply]
My apologies, it appears I've gotten in over my head, just disregard this.
